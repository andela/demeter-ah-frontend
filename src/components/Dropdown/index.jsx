import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import './index.scss';

const dropDown = ({ user, handleDropDown, handleSignOut }) => {
  const signOut = () => {
    handleSignOut();
  };

  const username = user && user.username;
  return (
    <div className="dropdown absolute max-w-85 min-w-48 flex flex-col">
      <div className="userinfo w-full py-2 bg-gray-30 text-center">
        <p className="username overflow-hidden p-2 text-gray-550 py-2">{`@${username}`}</p>
        <p className="fullname overflow-hidden p-2 text-gray-550 font-semibold whitespace-no-wrap">{`${user.firstName} ${user.lastName}`}</p>
      </div>
      <ul>
        <Link onClick={handleDropDown} to="/article/create">
          <li className="text-xs-2 border-b-2 border-gray-40 py-3 px-8 bg-white cursor-pointer text-center text-gray-550">
            Write a Story
          </li>
        </Link>
        <Link onClick={handleDropDown} to={`/profile/${username}/following`}>
          <li className="text-xs-2 border-b-2 border-gray-40 py-3 px-8 bg-white cursor-pointer text-center text-gray-550">
            Following
          </li>
        </Link>
        <Link onClick={handleDropDown} to={`/profile/${username}/articles`}>
          <li className="text-xs-2 border-b-2 border-gray-40 py-3 px-8 bg-white cursor-pointer text-center text-gray-550">
            Articles
          </li>
        </Link>
        <Link onClick={handleDropDown} to={`/profile/${username}/bookmark`}>
          <li className="text-xs-2 border-b-2 border-gray-40 py-3 px-8 bg-white cursor-pointer text-center text-gray-550">
            Bookmarks
          </li>
        </Link>
        <Link onClick={handleDropDown} to={`/profile/${username}/editprofile`}>
          <li className="text-xs-2 border-b-2 border-gray-40 py-3 px-8 bg-white cursor-pointer text-center text-gray-550">
            Profile
          </li>
        </Link>
        <Link onClick={handleDropDown} to={`/profile/${username}/stat`}>
          <li className="text-xs-2 border-b-2 border-gray-40 py-3 px-8 bg-white cursor-pointer text-center text-gray-550">
            Article Stats
          </li>
        </Link>
        <div onClick={signOut} role="button" onKeyPress={signOut} tabIndex={0}>
          <li className="text-xs-2 border-b-2 border-gray-40 py-3 px-8 bg-white cursor-pointer text-center text-gray-550">
            Sign out
          </li>
        </div>
      </ul>
    </div>
  );
};

const mapStateToProps = state => ({
  user: state.auth.user,
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(mapStateToProps)(dropDown);
