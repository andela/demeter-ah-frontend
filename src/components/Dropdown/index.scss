.dropdown {
  font-family: 'Montserrat', sans-serif;
  right: 0.2em;
  top: 130%;
  box-shadow: 1px 1px 10px 2px rgba(0, 0, 0, 0.2);
  li {
    &:hover {
      background-color: #fcfcfc;
      color: #474747;
    }
  }
}