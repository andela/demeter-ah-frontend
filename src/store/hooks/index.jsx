import { useEffect } from 'react';
import { getUser } from '../actions/signup';

export const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return [];
};
