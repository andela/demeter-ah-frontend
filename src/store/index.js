import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import combineReducers from './reducers';
import dummyAction from './actions/Signup';

const middlewares = applyMiddleware(thunk);

const store = createStore(
  combineReducers,
  composeWithDevTools(middlewares),
);

store.dispatch(dummyAction);

export default store;
