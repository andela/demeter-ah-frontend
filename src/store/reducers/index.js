import { combineReducers } from 'redux';
import authReducer from './Auth';
import resetPassword from './resetPassword';
import articles from './articles';
import editProfile from './editProfile';
import viewProfile from './viewProfile';

const reducers = combineReducers({
  auth: authReducer,
  resetPassword,
  articles,
  editProfile,
  viewProfile,
});

export default reducers;
