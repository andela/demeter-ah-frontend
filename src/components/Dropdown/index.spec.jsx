
import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from './index';
import store from '../../store';

describe('DropDown Component', () => {
  it('Should render without errors', () => {
    const component = shallow(<Dropdown store={store} user={{ username: 'kevo' }} />);
    expect(component).toMatchSnapshot();
  });
});
