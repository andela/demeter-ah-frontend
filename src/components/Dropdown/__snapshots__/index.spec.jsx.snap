// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown Component Should render without errors 1`] = `
<ContextProvider
  value={null}
>
  <dropDown
    dispatch={[Function]}
    isAuthenticated={false}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    user={Object {}}
  />
</ContextProvider>
`;
